packages/rs-dpp/src/document/extended_document.rs

```diff
@@ -68,6 +74,47 @@ pub struct ExtendedDocument {
     pub entropy: Bytes32,
 }
 
+impl Encode for ExtendedDocument {
+    fn encode<E: Encoder>(&self, encoder: &mut E) -> Result<(), EncodeError> {
```

Comment:
I'm not sure how it works - is the data being written to the encoder directly and then something like `encoder.finalize()` is called?

Reply:
Well yes, when you derive Encode it takes every attribute and encodes them and puts them back to back, but you don't need to derive, you can do things manually.

Reply:
I'm more interested in how the API works. For example, if you want to serialize something to a vec, you write something like `let encoded: Vec<u8> = bincode::serialize(&doc).unwrap();`, and then bincode figures out the encoder internally? I'm not really familiar with the barcode just yet, so I was wondering if my assumption is right

Reply:
This is not exactly a comment to this specific code, just figured that there's this place I can ask about it as well :)